### PR TITLE
fix: Convert MLSUM to fast

### PR DESCRIPTION
**Pull request by isaac-chung**
## Checklist for adding MMTEB dataset

Reason for dataset addition: Addresses #798 

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

File: docs/mmteb/points/865.jsonl [hunk 1]
```diff
@@ -0,0 +1,2 @@
+{"GitHub": "isaac-chung", "Bug fixes": 4}
+{"GitHub": "???", "Review PR": 2}
```
Comment (KennethEnevoldsen) [inline docs/mmteb/points/865.jsonl:2]:
```suggestion
{"GitHub": "KennethEnevoldsen", "Review PR": 2}
```

File: mteb/tasks/Clustering/multilingual/MLSUMClusteringP2P.py [hunk 2]
```diff
@@ -84,3 +86,88 @@ def dataset_transform(self, lang):
...
+            if is_solo.any():
+                split_ds = split_ds.select(np.nonzero(is_solo == False)[0])  # noqa: E712
+            lang_dict.update({split: split_ds})
+
+        self.dataset[lang] = self.stratified_subsampling(
+            DatasetDict(lang_dict),
+            self.seed,
+            self.metadata.eval_splits,
+            label="labels",
+            n_samples=2048,
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Clustering/multilingual/MLSUMClusteringP2P.py:172]: should probably be reflected in n_samples

File: mteb/tasks/Clustering/multilingual/MLSUMClusteringS2S.py [hunk 3]
```diff
@@ -79,3 +82,83 @@ def dataset_transform(self, lang):
...
+            if is_solo.any():
+                split_ds = split_ds.select(np.nonzero(is_solo == False)[0])  # noqa: E712
+            lang_dict.update({split: split_ds})
+
+        self.dataset[lang] = self.stratified_subsampling(
+            DatasetDict(lang_dict),
+            self.seed,
+            self.metadata.eval_splits,
+            label="labels",
+            n_samples=2048,
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Clustering/multilingual/MLSUMClusteringS2S.py:163]: same

File: results/intfloat__multilingual-e5-small/0a68dcd3dad5b4962a78daa930087728292b241d/MLSUMClusteringS2S.v2.json
Comment (KennethEnevoldsen) [inline results/intfloat__multilingual-e5-small/0a68dcd3dad5b4962a78daa930087728292b241d/MLSUMClusteringS2S.v2.json:1]: How do the scores compares to previous scores? (wondering if it is worth using more than 2048 samples)

**Review by KennethEnevoldsen — APPROVED** [review]

File: results/intfloat__multilingual-e5-small/0a68dcd3dad5b4962a78daa930087728292b241d/MLSUMClusteringS2S.v2.json
Comment (isaac-chung) [reply results/intfloat__multilingual-e5-small/0a68dcd3dad5b4962a78daa930087728292b241d/MLSUMClusteringS2S.v2.json:1]: Original (S2S) for french in `results/intfloat__multilingual-e5-small/MLSUMClusteringS2S.json`. They look quite similar.
```
"fr": {
      "main_score": 0.4283958792703536,
      "v_measure": 0.4283958792703536,
      "v_measure_std": 0.022410163550479215,
      "v_measures": [
        0.423138995462709,
        0.449800763047342,
        0.42903197222995726,
        0.4147259930495241,
        0.38087890904673305,
        0.4461669835484878,
        0.4563206424723956,
        0.45291153970332937,
        0.42103898526243955,
        0.40994400888061816
      ]
    }
```

**Pull request merged**